### PR TITLE
Drop the needless efi-cc and efi-ld parameters

### PR DESCRIPTION
**Pull request by superm1**
Fixes: #6

### CODE REVIEW

File: efi/meson.build [hunk 1]
```diff
@@ -159,7 +159,7 @@ else
   efi_format = ['--target=efi-app-@0@'.format(gnu_efi_arch)]
 endif
 
-libgcc_file_name = run_command(efi_cc, '-print-libgcc-file-name').stdout().strip()
+libgcc_file_name = run_command('gcc', '-print-libgcc-file-name').stdout().strip()
```
Comment (superm1) [inline efi/meson.build:162]: this one might need to stay `efi_cc`, will double check it.

File: efi/meson.build [hunk 2]
```diff
@@ -20,7 +20,7 @@ endif
 
 efi_libdir = get_option('efi-libdir')
 if efi_libdir == ''
-  cmd = 'cd /usr/lib/$(@0@ -print-multi-os-directory) && pwd'.format(efi_cc)
+  cmd = 'cd /usr/lib/$(@0@ -print-multi-os-directory) && pwd'.format('gcc')
```
Comment (superm1) [inline efi/meson.build:23]: at least for me this fails with my default CC command (`ccache cc`) if I keep it `efi_cc`.  Any other ideas welcome @rossburton.

**Comment by hughsie** [comment]
I get this on Fedora:

	FAILED: efi/fwup.so 
	/usr/bin/ld.bfd -o efi/fwup.so -T /home/hughsie/Code/fwupd-efi/efi/lds/elf_x86_64_efi.lds -shared -Bsymbolic -nostdlib -znocombreloc -L /usr/lib -L /usr/lib /usr/lib/crt0-efi-x86_64.o efi/fwupdate.o efi/fwup-debug.o efi/fwup-efi.o efi/fwup-common.o efi/fwup-sbat.o -lefi -lgnuefi /usr/lib/gcc/x86_64-redhat-linux/12/libgcc.a
	/usr/bin/ld.bfd: skipping incompatible /usr/lib/libefi.a when searching for -lefi
	/usr/bin/ld.bfd: skipping incompatible /usr/lib/libefi.a when searching for -lefi
	/usr/bin/ld.bfd: cannot find -lefi: file in wrong format
	/usr/bin/ld.bfd: skipping incompatible /usr/lib/libgnuefi.a when searching for -lgnuefi
	/usr/bin/ld.bfd: skipping incompatible /usr/lib/libgnuefi.a when searching for -lgnuefi
	/usr/bin/ld.bfd: cannot find -lgnuefi: file in wrong format

**Comment by superm1** [comment]
Hmm I guess on Fedora it's using a different linker by default?  If you switch to `ld.gold` does it work?

**Comment by superm1** [comment]
I think the problem is actually the RPM packaging doing something super weird.  I think that `/usr/lib/gnuefi.a` doesn't match the right architecture.  We can probably work around it, but can you check that?

**Comment by superm1** [comment]
> /usr/lib/gnuefi/libgnuefi.a -> ia32/libgnuefi.a

That's the problem.  It should be a symlink to the x64 version.  I think we can do a workaround in fwupd-efi, but this should be fixed in the RPM.

**Comment by superm1** [comment]
Have a try with what I just pushed

File: efi/meson.build [hunk 3]
```diff
@@ -159,36 +153,36 @@ else
...
+                        command : [cc.cmd_array(), '-c', '@INPUT@', '-o', '@OUTPUT@']
                                   + compile_args)
 
 o_file5 = custom_target('fwup-sbat.o',
                         output : 'fwup-sbat.o',
                         command : [
                           generate_sbat,
                           '@OUTPUT@',
-                          '--cc', efi_cc,
+                          '--cc', '@0@'.format(' '.join(cc.cmd_array())),
```
Comment (hughsie) [inline efi/meson.build:185]: do you need the @0@ or can you use `' '.join(cc.cmd_array())` here?

File: efi/meson.build [hunk 4]
```diff
@@ -20,11 +18,7 @@ endif
 
 efi_libdir = get_option('efi-libdir')
 if efi_libdir == ''
-  cmd = 'cd /usr/lib/$(@0@ -print-multi-os-directory) && pwd'.format(efi_cc)
-  ret = run_command('sh', '-c', cmd)
-  if ret.returncode() == 0
-    efi_libdir = ret.stdout().strip()
-  endif
+  efi_libdir = '/usr/lib/' / run_command(cc.cmd_array(), '-print-multi-os-directory').stdout().strip()
```
Comment (hughsie) [inline efi/meson.build:21]: I know it's valis but the naked `/` trips me up every time. Would join_paths be clearer here?

**Comment by hughsie** [comment]
> Have a try with what I just pushed

Works.

File: efi/meson.build (hunk 4 above)
Comment (superm1) [reply efi/meson.build:21]: `/` is the one you're supposed to use for meson OS independent code, but I guess this really is Linux only the way it's done right now.

File: efi/meson.build (hunk 3 above)
Comment (superm1) [reply efi/meson.build]: fixed

**Comment by hughsie** [comment]
I guess my only concern is:

	Message: efi-libdir: "/usr/lib/../lib64"
	Message: efi-ldsdir: "/home/hughsie/Code/fwupd-efi/efi/lds"
	Message: efi-crtdir: "/usr/lib/../lib64/gnuefi"

which I *think* is fine?

**Comment by rossburton** [comment]
I can test this on an aarcch64 machine on Monday if nobody else has access.

**Comment by superm1** [comment]
> I can test this on an aarcch64 machine on Monday if nobody else has access.

Yes, please!

**Comment by superm1** [comment]
> I guess my only concern is:
> 
> 	Message: efi-libdir: "/usr/lib/../lib64"
> 	Message: efi-ldsdir: "/home/hughsie/Code/fwupd-efi/efi/lds"
> 	Message: efi-crtdir: "/usr/lib/../lib64/gnuefi"
> 
> which I *think* is fine?

I thought it should be fine too.

**Pull request merged**